test/coordinator_synthetic.py

```
@@ -0,0 +1,1043 @@
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            timeout=5,
+            check=False,
+        )
+
+        self.assertEqual(result.returncode, 1)
+        self.assertIn("connect failed", result.stderr)
+        self.assertNotIn("invalid port", result.stderr)
+
+    def run_command(self, *args):
+        return subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)